FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/workmanager/HandlePasswordProtectedMsgWorker.kt

```diff
@@ -0,0 +1,425 @@
+              val fromAddress = mimeMsgWithAttachments.getFromAddress()
+              val domain = EmailUtil.getDomain(fromAddress)
+              val idToken = getGoogleIdToken()
+              val replyToken = fetchReplyToken(apiRepository, domain, idToken)
+              val messageUploadRequest = MessageUploadRequest(
+                associateReplyToken = replyToken,
+                from = fromAddress,
+                to = toCandidates.map { (it as InternetAddress).address },
+                cc = ccCandidates.map { (it as InternetAddress).address },
+                bcc = bccCandidates.map { (it as InternetAddress).address }
+              )
+
+              //prepare bodyWithReplyToken
+              val replyInfo = Base64.getEncoder().encodeToString(
+                GsonBuilder().create().toJson(messageUploadRequest).toByteArray()
+              )
+
+              val infoDiv = genInfoDiv(replyInfo)
+              val originalText = getDecryptedContentFromMessage(
+                mimeMsgWithAttachments,
+                accountSecretKeys,
+                keysStorage
+              )
+              val bodyWithReplyToken = originalText + "\n\n" + infoDiv
+
+              // construct a regular plain mime message using bodyWithReplyToken + attachments
+              mimeMsgWithAttachments.setRecipients(Message.RecipientType.TO, toCandidates)
+              mimeMsgWithAttachments.setRecipients(Message.RecipientType.CC, ccCandidates)
+              mimeMsgWithAttachments.setRecipients(Message.RecipientType.BCC, bccCandidates)
+              val multipart = mimeMsgWithAttachments.content as Multipart
+              //need to remove 'encrypted.asc' from the existing MIME
+              multipart.removeBodyPart(0)
+              multipart.addBodyPart(MimeBodyPart().apply { setText(bodyWithReplyToken) }, 0)
+              mimeMsgWithAttachments.saveChanges()
+
+              //prepare raw MIME
+              val rawMimeMsg = String(ByteArrayOutputStream().apply {
+                mimeMsgWithAttachments.writeTo(this)
+              }.toByteArray())
+
+              //encrypt the raw MIME message ONLY FOR THE MESSAGE PASSWORD
+              val pwdEncryptedWithAttachments = PgpEncryptAndOrSign.encryptAndOrSignMsg(
+                msg = rawMimeMsg,
+                pubKeys = emptyList(),
+                prvKeys = emptyList(),
+                passphrase = Passphrase.fromPassword(
+                  KeyStoreCryptoManager.decryptSuspend(String(requireNotNull(msgEntity.password)))
+                )
+              )
+
+              //upload resulting data to FES
+              val fesUrl = uploadMsgToFESAndReturnUrl(
+                apiRepository,
+                domain,
+                idToken,
+                messageUploadRequest,
+                pwdEncryptedWithAttachments
+              )
+
+              updateExistingMimeMsgWithUrl(msgEntity, fesUrl)
+            } else {
+              //ask Tom about that
```

Comment:
@tomholub also, please pay attention to this line.

Reply:
I'm not sure why should there be an IF above. I'd probably remove it, which would also remove the ELSE.

Reply:
Can we send message via FES if `toCandidates` is empty? Or can I use the following?

```
if (toCandidates.isNotEmpty() || ccCandidates.isNotEmpty() || bccCandidates.isNotEmpty()) {
     //send via FES
}
```

Reply:
@tomholub Btw, FES now doesn't check presence of `to` recepients. Is that correct? See https://github.com/FlowCrypt/flowcrypt-enterprise-on-prem/blob/master/fes-lib/src/main/kotlin/com/flowcrypt/fes/server/handlers/api/portal/MessageUploadPost.kt

Reply:
> Can we send message via FES if toCandidates is empty? Or can I use the following?

When would that ever happen?

> @tomholub Btw, FES now doesn't check presence of to recepients. (Actually, it doesn't check presence of any recepients, but we normally consider to as mandatory.) Is that correct? See https://github.com/FlowCrypt/flowcrypt-enterprise-on-prem/blob/master/fes-lib/src/main/kotlin/com/flowcrypt/fes/server/handlers/api/portal/MessageUploadPost.kt I think FES should check this, and apps & browser ext should not allow sending w/o at least single to.

I think it's running an old version that didn't require it. I'll make a separate issue to check once updated.

Reply:
> When would that ever happen?

 What about this one?

![Screenshot_2022-01-04-12-09-10-69_8a5443484bc9afa2d32cf276c137039c](https://user-images.githubusercontent.com/2863246/148043310-dee3c402-413f-47ad-8a93-b012d96ec824.jpg)


We have `cc` here

Reply:
Yes, of course that is valid, and can be sent. What I meant is, why do you need that IF at all? When would it ever happen that you are in FES upload worker, and have a message where you have zero non-pgp recipients?

Reply:
I'm confused a little. Let's me clarify some things.

1. We send to FES a MIME message with original recipients or only with non-pgp recipients?

> When would it ever happen that you are in FES upload worker, and have a message where you have zero non-pgp recipients?

2. There should not be such a case. As I understand it doesn't metter if we have empty `to`, `cc` or `bcc`(if we are talking about `https://fes.{domain}/api/v1/message`). We should have at least one recipient(`to`, `cc` or `bcc`). Right?

Reply:
Accoridng to what I've seen in the code, currently FES  should happily accept message even w/o any receipients.

Reply:
> I'm confused a little. Let's me clarify some things.
> 
>     1. We send to FES a MIME message with original recipients or only with non-pgp recipients?

It should include the original recipients, all of them, in the request.

> 
> > When would it ever happen that you are in FES upload worker, and have a message where you have zero non-pgp recipients?
> 
>     2. There should not be such a case. 

That's what I thought. That's why the if/else may not need to be there at all.

> As I understand it doesn't metter if we have empty `to`, `cc` or `bcc`(if we are talking about `https://fes.{domain}/api/v1/message`). We should have at least one recipient(`to`, `cc` or `bcc`). Right?

Currently FES will allow empty, but we'll change that to require that at least one of the three is present. But since you will be adding all recipients, and there will always be some, then this doesn't matter - can remove the if.